keps/sig-network/2943-networkpolicy-status/README.md

```diff
@@ -0,0 +1,462 @@
+
+### Graduation Criteria
+
+#### Alpha
+
+- Add a feature gated new status subresource to NetworkPolicy
+- Initial e2e tests completed
+- Communicate Network Policy providers about the new field
+- Add validation tests in API
+
+#### Beta
+
+- Status has been supported for at least 1 minor release
+- **two** commonly used NetworkPolicy providers implement the new field,
+with generally positive feedback on its usage.
+- Feature Gate is enabled by Default.
+
+#### GA
+
+- At least **three** NetworkPolicy providers support the status subresource and the conditions
+- Status has been enabled by default for at least 2 minor release
+
+### Upgrade / Downgrade Strategy
+
+If upgraded no impact should happen as this is a new field.
+
+If downgraded the Network Policy provider wont be able to write into 
+this new field, which may lead to errors and inconsistency.
+
+The provider implementing this feature should deal with errors 
+returned from the API Server and bypass them in case of the field not existing in the current version.
+
+### Version Skew Strategy
+
+N/A
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+-->
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: NetworkPolicyStatus
+  - Components depending on the feature gate:
+    - API Server
+    - Network Policy Providers
+
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+ Yes. Rolling back after Status has been added will keep 
+ the status persisted on ETCD but the field wont be accessible on the API Server for GET/PUT operations.
+
+Network Policy Providers wont be able to update the status field and 
+this may lead to this providers returning errors when trying to 
+access / change this field.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Network Policy Providers will be able again to add the status conditions.
+###### Are there any tests for feature enablement/disablement?
+
+Not right now, but this can be targeted to Beta as soon as we have an initial implementation
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollout won't cause any problem, as this is a new field.
+
+A rollback can be problematic from the Network Policy Provider perspective, 
+as if decided to use this field, the provider will have to deal with 
+failures if the Feature Gate is disabled or rolled back and the provider 
+already relies on this field to provide its status.
+
+###### What specific metrics should inform a rollback?
+
+  The increase of 5xx http error count on Network Policies Status subresource Endpoint
```

Comment:
Maybe mention the metric name explicitly:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go#L76

[We're splitting by code there, so you can get the number you need]